### PR TITLE
howl: init at 0.5.3

### PR DESCRIPTION
**Pull request by pacien**
###### Motivation for this change

This PR adds a new package for the Howl text editors and registers myself as a maintainer.

The package builds on my NixOS machine and the programs runs fine.

Since the introduction of this new package should not break anything, could this be back-ported to `release-18.03` and `release-18.09`?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by timokau — CHANGES_REQUESTED** [review]
Great to see the maintainers list grow, welcome :tada: 

Here are a few nitpicks.

File: pkgs/applications/editors/howl/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl, makeWrapper, pkgconfig, gtk3, librsvg }:
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "howl-${version}";
+  version = "0.5.3";
+
+  src = fetchurl {
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/applications/editors/howl/default.nix]: I use a release tarball containing the pre-downloaded dependencies sources.
Checking out from git would require downloading those manually.

File: pkgs/applications/editors/howl/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    url = "https://github.com/howl-editor/howl/releases/download/0.5.3/howl-0.5.3.tgz";
+    sha256 = "0gnc8vr5h8mwapbcqc1zr9la62rb633awyqgy8q7pwjpiy85a03v";
+  };
+
+  sourceRoot = "./${name}/src";
+  installFlags = [ "PREFIX=$(out)" ];
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/applications/editors/howl/default.nix]: The Makefile sets the PREFIX value to "/usr/local" if not explicitly overridden: https://github.com/howl-editor/howl/blob/0.5.3/src/Makefile#L1

I added a comment.

File: pkgs/applications/editors/howl/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ makeWrapper gtk3 librsvg ];
+  enableParallelBuilding = true;
+
+  postInstall = ''
+    wrapProgram $out/bin/howl \
+      --set GDK_PIXBUF_MODULE_FILE "$GDK_PIXBUF_MODULE_FILE"
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/applications/editors/howl/default.nix]: I added the comment.

File: pkgs/applications/editors/howl/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchurl {
+    url = "https://github.com/howl-editor/howl/releases/download/0.5.3/howl-0.5.3.tgz";
+    sha256 = "0gnc8vr5h8mwapbcqc1zr9la62rb633awyqgy8q7pwjpiy85a03v";
+  };
+
+  sourceRoot = "./${name}/src";
+  installFlags = [ "PREFIX=$(out)" ];
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ makeWrapper gtk3 librsvg ];
+  enableParallelBuilding = true;
```
[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/applications/editors/howl/default.nix]: There's no test provided.

**Comment by pacien** [comment]
@timokau Thanks for your feedback.
I've replied to the issues you have pointed out and made some changes.

File: pkgs/applications/editors/howl/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/applications/editors/howl/default.nix]: What dependencies? The README suggests building from a clone:

> Clone the repository or download and unpack a release. Cd into the src directory, and run make && sudo make install. The installed binary will be named howl.

[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/applications/editors/howl/default.nix]: When cloning from git, those dependencies (LUAJIT and LPEG) are downloaded by the Makefile: https://github.com/howl-editor/howl/blob/0.5.3/src/Makefile#L60

[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/applications/editors/howl/default.nix]: I see. Please also add a comment explaining that to the `fetchurl`, then this should be good to go.

[the remark this answers is not in the document]
Comment (pacien) [reply pkgs/applications/editors/howl/default.nix]: I added it but GitHub doesn't show it on the PR page: https://github.com/NixOS/nixpkgs/blob/424c0009519287cc16f048888e35a5fe4e825640/pkgs/applications/editors/howl/default.nix#L9

**Comment by timokau** [comment]
@GrahamcOfBorg build howl

**Review by timokau — APPROVED** [review]

**Comment by pacien** [comment]
It seems like darwin isn't supported.
How can it be excluded from the build platforms?

**Comment by timokau** [comment]
If the aarch64 build succeeds (apparently its still running), you can replace `platforms.all` by `platforms.linux`.

**Comment by timokau** [comment]
Since it fails on aarch64 too, you set `platforms = [ "i686-linux" "x86_64-linux" ]`.

**Comment by pacien** [comment]
The `platforms` variable has been set accordingly.

**Comment by timokau** [comment]
@GrahamcOfBorg build howl

**Comment by timokau** [comment]
Great, thank you for your patience with the review :)

**Comment by timokau** [comment]
I agree, I don't think there even is a hard policy on that. There probably should be. @samueldr what do you think on backporting new packages?

**Comment by samueldr** [comment]
The current backport policies are about keeping the existing software set bug-free and safe security wise. We don't have policies about new software in stable, but current unwritten agreements are that it's not really done.

While it wouldn't break anything, Nix, Nixpkgs and NixOS makes composition easy enough that "leaf derivations" like this one should be trivial to either add locally to an overlay, or use from unstable.

**Pull request merged**